app/controllers/users_controller.rb

```diff
@@ -12,21 +12,34 @@ def create
       flash[:success] = "Welcome, #{@user.name}! You are now logged in and registered."
       redirect_to profile_path
     else
-      flash[:error] = @user.errors.full_messages.uniq.to_sentence
+      flash.now[:error] = @user.errors.full_messages.uniq.to_sentence
       render :new
     end
   end
 
-  def edit
+  def edit_profile
   end
 
-  def update
+  def edit_password
+  end
+
+  def update_profile
     if current_user.update(update_profile_params)
       flash[:success] = 'Your profile data has been updated!'
       redirect_to profile_path
     else
-      flash[:error] = @user.errors.full_messages.uniq.to_sentence
-      render :edit
+      flash.now[:error] = current_user.errors.full_messages.uniq.to_sentence
+      render :edit_profile
+    end
+  end
+
+  def update_password
+    if current_user.update(update_password_params)
+      flash[:success] = 'Your password has been updated successfully!'
+      redirect_to profile_path
+    else
+      flash.now[:error] = current_user.errors.full_messages.uniq.to_sentence
+      render :edit_password
```

Comment:
A lot of duplicated functionality here, but we'll refactor this later.

Reply:
This has been added as an issue (#83)